### PR TITLE
kernel: Reorder config files 

### PR DESCRIPTION
**Pull request by hauke**
* mediatek: Remove double CONFIG_HARDEN_BRANCH_PREDICTOR
    
    CONFIG_HARDEN_BRANCH_PREDICTOR option is included two times in this
    configuration file. Remove one definition. On arm32 SoC it should not be
    needed.
    
    Fixes: 54f9744c820a ("treewide: disable spectre mitigation on unaffected Arm32 targets")

 * kernel: Reorder config files
    
    Reorder the kernel configuration files.
    
    I used this script to generate the commit, llama3.2 helped with the
    script:
    ```
    
    find_files=$(find target -type f -name 'config-*' -print)
    
    if [ -n "$find_files" ]; then
      echo "Found files:"
      for file in $find_files; do
        echo "Running kconfig.pl for $file"
        LANG=C ./scripts/kconfig.pl '+' "$file" /dev/null | sponge "$file"
      done
    else
      echo "No files named config-6.6 found."
    fi
    ```

### CODE REVIEW

**Review by neheb — APPROVED** [review]

File: target/linux/generic/config-filter [hunk 1]
```diff
@@ -1,17 +1,17 @@
+# CONFIG_ARCH_(ENABLE|HAS|HAVE|INLINE|SUPPORTS|USE|WANT|STACKWALK)_.* is not set
```
Comment (neheb) [inline target/linux/generic/config-filter:1]: is this right?

Comment (hauke) [reply target/linux/generic/config-filter:1]: I think it is no problem to move this.

**Comment by Ansuel** [comment]
@hauke I also faced recently the task of cleaning a kernel config... Worth adding one of these utility script in maintainer-tools or scripts?

**Comment by hauke** [comment]
I added the script in a separate commit.

**Comment by Ansuel** [comment]
Thanks a lot very useful!

**Review by neheb — APPROVED** [review]

**Pull request merged**